### DISCUSSION TITLE
Filter elements in other dataframe

### DISCUSSION THREAD

**Discussion by romikforest** [vaexio/vaex#1669] [Q&A] [answered]
Hi.
Is there a simple way to filter elements from one dataframe on condition on another, basically supposed to be with the same indices/positions in this case:

```
print(questions[questions.inf == question])  # This works and filter the expected records
print(answers[questions.inf == question])  # But this is empty. answers df have the same len as questions
```

Every dataframe has only one column `inf`.
I can do something like:

`result = answers.take(np.nonzero((questions.inf == question).to_numpy())[0]).inf.tolist()`

But I'm not sure it is efficient at all.

Also I tried to use a combined dataframe, but it uses nest_asyncio after that. nest_asyncio uses python only implementation of asyncio loop, incompatible with uvloop, and it's not unacceptable for my applications.

**Comment by maartenbreddels** [comment, marked as answer]
Hi,

we should warn that this is mixing dataframes, which happens to work because the names of the dataframes are the same. You can do a join, without a key, which will join based on row number (which takes 0 time), and then filter.

Does this help?

cheers,

Maarten

> combined dataframe

What is this?

I'm thinking about avoiding nest_asyncio, see https://github.com/vaexio/vaex/pull/1546 and https://github.com/spyder-ide/spyder/issues/16183 for discussions on this.
